index.js

```diff
@@ -208,6 +208,13 @@ module.exports = function leafletImage(map, callback) {
             });
         };
 
+        if (useAjax) {
+            getBase64(url, function(data) {
+                im.src = 'data: image/png;base64, ' + data;
+                im.onload();
+            });
+            return;
+        }
         im.src = url;
```

Comment:
I corrected the else to me more explicit.  Also I fixed a bug that caused onload to be called multiple times depending on latency and I also now cache the images internally to cut down on requests.